### PR TITLE
Introduce NewClusterModal

### PR DESCRIPTION
**Pull request by jtomasek**
This allows setting cluster name and openshiftVersion prior to cluster creation

Screenshot 2020-05-05 at 14 15 25

### CODE REVIEW

File: src/components/clusters/Clusters.tsx [hunk 1]
```diff
@@ -83,12 +83,7 @@ const Clusters: React.FC = () => {
             <ClustersTable rows={clusterRows} deleteCluster={deleteCluster} />
           </PageSection>
           <ClusterWizardToolbar>
-            <ToolbarButton
-              variant={ButtonVariant.primary}
-              component={(props) => <Link to="/clusters/new" {...props} />}
-            >
-              Create New Cluster
-            </ToolbarButton>
+            <NewClusterModalButton ButtonComponent={ToolbarButton} />
```
Comment (mareklibra) [inline src/components/clusters/Clusters.tsx:86]: Why do we need this flexibility with button component?

**Comment by mareklibra** [comment]
We should not allow the user to create a new cluster with conflicting name (of an existing cluster). Opening a jira ticket for that.

File: src/components/clusters/Clusters.tsx (hunk 1 above)
Comment (jtomasek) [reply src/components/clusters/Clusters.tsx:86]: You're right, it is not needed here in this case (I've copied it over from discovery Iso modal which is triggered from 2 different buttons.

**Comment by jtomasek** [comment]
> We should not allow the user to create a new cluster with conflicting name (of an existing cluster).

I'll fix this as part of https://github.com/openshift-metal3/facet/pull/87

**Pull request merged**